ttnn/cpp/ttnn/operations/ccl/all_gather/all_gather_pybind.cpp

```diff
@@ -61,6 +85,12 @@ void py_bind_all_gather(pybind11::module& module) {
         Args:
             * :attr:`input_tensor` (ttnn.Tensor): multi-device tensor
             * :attr:`dim` (int)
+            * Following are applicable only for Linear Topology
+            * :attr:`cluster_axis` (int):
+                Provided a MeshTensor, the axis corresponding to MeshDevice
```

Comment:
Can we provide a link to the mesh tensor programming guide (it's linked to in the main project readme)